src/Nethermind/Nethermind.Vault/JsonRpc/VaultModule.cs

```diff
@@ -0,0 +1,201 @@
+            } 
+            catch (Exception e) 
+            {
+                return ResultWrapper<object>.Fail(e.ToString());
+            }
+        }
+
+        public async Task<ResultWrapper<object>> vault_listKeys(string vaultId)
+        {
+            try 
+            {
+                var args = new Dictionary<string, object> {};
+                var result = await _initVault.ListVaultKeys(_vaultConfig.Token, vaultId, args);   
+
+                return ReturnResult(result);
+            } 
+            catch (Exception e) 
+            {
+                return ResultWrapper<object>.Fail(e.ToString());
+            }
+        }
+
+        public async Task<ResultWrapper<object>> vault_listSecrets(string vaultId)
+        {
+            try 
+            {
+                var args = new Dictionary<string, object> {};
+                var result = await _initVault.ListVaultSecrets(_vaultConfig.Token, vaultId, args);   
+
+                return ReturnResult(result);
+            } 
+            catch (Exception e) 
+            {
+                return ResultWrapper<object>.Fail(e.ToString());
+            }
+        }
+
+        public async Task<ResultWrapper<object>> vault_listVaults()
+        {
+            try 
+            {
+                var args = new Dictionary<string, object> {};
+                var result = await _initVault.ListVaults(_vaultConfig.Token, args);
+
+                return ReturnResult(result);
+            } 
+            catch (Exception e) 
+            {
+                return ResultWrapper<object>.Fail(e.ToString());
+            }
+        }
+
+        public async Task<ResultWrapper<object>> vault_signMessage(string vaultId, string keyId, string message)
+        {
+            try 
+            {
+                var args = new Dictionary<string, object> {};
+                var result = await _initVault.SignMessage(_vaultConfig.Token, vaultId, keyId, message);   
+
+                return ReturnResult(result);
+            } 
+            catch (Exception e) 
+            {
+                return ResultWrapper<object>.Fail(e.ToString());
+            }
+        }
+
+        public async Task<ResultWrapper<object>> vault_verifySignature(string vaultId, string keyId, string message, string signature)
+        {
+            try 
+            {
+                var args = new Dictionary<string, object> {};
+                var result = await _initVault.VerifySignature(_vaultConfig.Token, vaultId, keyId, message, signature);   
+
+                return ReturnResult(result);
+            } 
+            catch (Exception e) 
+            {
+                return ResultWrapper<object>.Fail(e.ToString());
+            }
+        }
+
+        private ResultWrapper<object> ReturnResult((int, string) result)
+        {
+            if (result.Item1 == 200 || result.Item1 == 201)
+            {
+                dynamic jsonResult  = JsonConvert.DeserializeObject(result.Item2);
```

Comment:
dynamic is super slow - are we sure it will not cause additional DLR packages loading? it may be better to avoid it and just unpack to a DAO type

Reply:
hmmm
hard to say